File: src/main/java/com/hospital/management/bed/service/BedService.java
package com.hospital.management.bed.service;

import com.hospital.management.bed.dto.BedRequestDTO;
import com.hospital.management.bed.dto.BedResponseDTO;

import java.util.List;

public interface BedService {

    BedResponseDTO saveBed(BedRequestDTO dto);

    List<BedResponseDTO> getAllBeds();

    BedResponseDTO getBedById(Long id);

    BedResponseDTO updateBed(Long id, BedRequestDTO dto);

    void deleteBed(Long id);
}

File: src/services/bedService.js
import api from "./api";

export const getAllBeds = () => {
  return api.get("/api/beds");
};

export const getBedById = (id) => {
  return api.get(`/api/beds/${id}`);
};

export const saveBed = (data) => {
  return api.post("/api/beds", data);
};

export const updateBed = (id, data) => {
  return api.put(`/api/beds/${id}`, data);
};

export const deleteBed = (id) => {
  return api.delete(`/api/beds/${id}`);
};